control/freqplot.py

```diff
@@ -117,65 +117,100 @@ def bode_plot(syslist, omega=None, dB=None, Hz=None, deg=None,
     # If argument was a singleton, turn it into a list
     if (not getattr(syslist, '__iter__', False)):
         syslist = (syslist,)
-
-    mags, phases, omegas = [], [], []
+    
+    if omega is None:
+        # Select a default range if none is provided
+        omega = default_frequency_range(syslist, Hz=Hz)
+    elif (isinstance(omega, tuple) or isinstance(omega, list)) and len(omega) == 2:
```

Comment:
Thanks for all the valuable comments. Great to get some feedback.

That is how I interpreted the original documentation but the feature was not implemented.
The more I think about the less I like it.
I will implement an additional argument `omega_limits`. That should be more clear.

I agree the frequency-setup stuff should be refactored.